extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -145,6 +153,28 @@ protected synchronized TaskStatus run(Job job, long launchTimeout, long timeout)
     }
   }
 
+  private void writeTaskPayload(Task task)
+  {
+    try {
+      Path file = Files.createTempFile(taskId.getOriginalTaskId(), "task.json");
+      try {
```

Comment:
The nested try-catch seems can be simplified.

Reply:
how do you reccommend simplifying this? the createTempFile can throw a IOException

Reply:
Don't think we need to handle the IOException explicitly as it is wrapped to RuntimeException anyway.
So the code can be like:
```
      Path file = null;
      try {
        file = Files.createTempFile(taskId.getOriginalTaskId(), "task.json");
        FileUtils.writeStringToFile(file.toFile(), mapper.writeValueAsString(task), Charset.defaultCharset());
        taskLogs.pushTaskPayload(task.getId(), file.toFile());
      }
      catch (Exception e) {
        log.error("Failed to write task payload for task: %s", taskId.getOriginalTaskId());
        throw new RuntimeException(e);
      }
      finally {
        if (file ! = null) {
          Files.deleteIfExists(file);
        }
      }
    }
```

Reply:
the issue is the deleteIfExists call throws a IOException so I didn't want to have the whole function do that